### PR TITLE
Use Windows SSH agents for CI

### PR DESCRIPTION
**Pull request by jzila**

### CODE REVIEW

**Comment by jzila** [comment]
CI failure is a broken test on master

**Comment by jzila** [comment]
@akalin-keybase, @zanderz: this is ready for review. We will now be using SSH-connected Windows agents, which are powered by cygwin. This gives us the ability to use an auto-provisioned spot fleet based on demand for Windows builds. It should be cheaper and work more smoothly. I've also upgraded the instance types to the newest, fastest ones.

**Comment by jzila** [comment]
I'm going to merge because of the huge benefits. I will implement any feedback in a subsequent PR.

File: Jenkinsfile [hunk 1]
```diff
@@ -284,31 +287,22 @@ def testGo(prefix) {
...
-            dirs = getTestDirsNix()
-            slash = '/'
-            goversion = sh(returnStdout: true, script: "go version").trim()
-            // Ideally, we'd do this on Windows, too, but we'd have to figure
-            // out how to do it with batch files or PowerShell.
-            retry(5) {
-                sh 'go get -u github.com/golang/lint/golint'
-            }
-            sh 'make -s lint'
+            // Windows `gofmt` pukes on CRLF, so only run on *nix.
```
Comment (akalin-keybase) [inline Jenkinsfile]: sad...

**Pull request merged**